cryptoki/src/session/object_management.rs

```diff
@@ -14,6 +14,42 @@ const MAX_OBJECT_COUNT: usize = 10;
 
 impl Session {
     /// Search for session objects matching a template
+    ///
+    /// # Arguments
+    /// * `template` - A [Attribute] of search parameters that will be used
+    ///                 to find objects.
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// use cryptoki::session::Session;
+    /// use cryptoki::context::Pkcs11;
+    /// use cryptoki::object::{Attribute, AttributeType, CertificateType, ObjectClass, ObjectHandle};
+    ///
+    /// let mut client = Pkcs11::new(
+    ///     std::env::var("PKCS11_SOFTHSM2_MODULE")
+    ///        .unwrap_or_else(|_| "/usr/local/lib/softhsm/libsofthsm2.so".to_string()),
+    /// ).expect("Failed to load library");
+    /// client.initialize(cryptoki::context::CInitializeArgs::OsThreads).expect("Failed to init session");
+    ///
+    /// // Use the first slot
+    /// let slots = client.get_all_slots().expect("Failed to list slots on card");
+    /// let slot = slots.first().expect("Failed to find 1st slot");
+    /// let session = client.open_ro_session(*slot).expect("Failed to open read-only session");
```

Comment:
I wonder if we need to make this visible as part of this example, as setting up a session is common to the vast majority of operations. Instead we could maybe point in a comment to a different place where this is shown.

Reply:
Hmm yeah I agree that since most of the library is basically without examples it would be good to copy them to other places (say [`Pkcs11::open_ro_session`](https://docs.rs/cryptoki/latest/cryptoki/context/struct.Pkcs11.html#method.open_ro_session)) and hide them here (using `#` prefix in code examples) and maybe link to it.

This is more work so I'm not sure if you'd like to do it. As it is the example is already good I think.